src/v/datalake/coordinator/iceberg_file_committer.cc

```diff
@@ -103,8 +103,13 @@ iceberg_file_committer::commit_topic_files_to_catalog(
     if (tp_it == state.topic_to_state.end()) {
         co_return chunked_vector<mark_files_committed_update>{};
     }
+    if (
+      tp_it->second.lifecycle_state == topic_state::lifecycle_state_t::purged) {
+        co_return chunked_vector<mark_files_committed_update>{};
+    }
+
```

Comment:
Since we're making replicated decisions about state below, we should plumb an expected revision id here too

Reply:
The revision check is kind of useless here, as the only place where the caller can get the expected revision is the stm state itself. Basically the semantics are "Just commit whatever you currently have in the stm state on that topic", we don't need a revision check for that. This conditional by contrast is just an optimization to avoid loading the table.

Reply:
Saving the revision id at the start to check that it hasn't change later probably makes sense though.